src/insertion.jl

```diff
@@ -191,6 +190,23 @@ function set!(indices::AbstractIndices{I}, i1::I, i2::I) where {I}
     end
 end
 
+"""
+    setwith!(f, dict::AbstractDictionary, i, value)
+
+Update the value at `i` with the function `f` (`f(dict[i], value)`) or insert `value`.
+
+Hint: Use [`mergewith!`](@ref) to exclusively update an existing value, and `insert!` to exclusively
```

Comment:
Looks like `getindex` would be a better fit than `mergewith!` in this docstring, given that you also mention `insert!`.

Reply:
Haha ups bad copy paste